sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java

```diff
@@ -720,6 +722,9 @@ private CoderFactory getDefaultCoderFactory(Class<?> clazz) throws CannotProvide
       return getDefaultCoder(clazz);
     } else if (type instanceof ParameterizedType) {
       return getDefaultCoder((ParameterizedType) type, typeCoderBindings);
+    } else if (type instanceof TypeVariable
+        && TypeToken.of(type).isSubtypeOf(Serializable.class)) {
+      return SerializableCoder.PROVIDER.getCoder(TypeDescriptor.of(type));
```

Comment:
This is a bit overspecialized. What I think makes sense is get the `TypeDescriptor` to the `fallbackCoderProvider` on [line 336](https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java#L336). I think you just need to adjust the overload that is used on [line 722](https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderRegistry.java#L722) - it is just a historical accident that it only accepts a `Class`. (you might need to adjust names and `public`/`private`, etc)